core/node/state_keeper/src/testonly/test_batch_executor.rs

```diff
@@ -756,15 +756,15 @@ impl StateKeeperIO for TestIO {
         Ok(())
     }
 
-    async fn reject(&mut self, tx: &Transaction, error: &str) -> anyhow::Result<()> {
+    async fn reject(&mut self, tx: &Transaction, error: UnexecutableReason) -> anyhow::Result<()> {
         let action = self.pop_next_item("reject");
         let ScenarioItem::Reject(_, expected_tx, expected_err) = action else {
             panic!("Unexpected action: {:?}", action);
         };
         assert_eq!(tx, &expected_tx, "Incorrect transaction has been rejected");
         if let Some(expected_err) = expected_err {
```

Comment:
why this will be better?

Reply:
Should reiterate again that this makes most sense if `UnexecutableReason` has well-defined granular variants. In this case, comparing `UnexecutableReason`s is better because it's more strongly typed, meaning:

- More difficult to screw up (you can't accidentally mistype an expected error message; it cannot get unnoticeably outdated etc.)
- Easier to read the intention behind the test case
- Easier to debug in case of an error

Reply:
understood and accepted
thanks